test/integration.test.js

```diff
@@ -29,17 +29,20 @@ const npmRegistry = require('./helpers/npm-registry');
 const requireNoCache = proxyquire.noPreserveCache();
 
 // Environment variables used with semantic-release cli (similar to what a user would setup)
+const {GITHUB_ACTION, GITHUB_TOKEN, ...processEnvWithoutGitHubActionsVariables} = process.env;
 const env = {
+  ...processEnvWithoutGitHubActionsVariables,
   ...npmRegistry.authEnv,
+  CI: 'true',
   GH_TOKEN: gitbox.gitCredential,
-  GITHUB_URL: mockServer.url,
   TRAVIS: 'true',
-  CI: 'true',
   TRAVIS_BRANCH: 'master',
   TRAVIS_PULL_REQUEST: 'false',
+  GITHUB_API_URL: mockServer.url,
 };
```

Comment:
The main problem why the CI tests started to fail after migrating from Travis to GitHub Actions was that the environment variables set by the GitHub Actions environment sneaked into `process.env`. I always thought that setting the [`env` option for exec](https://github.com/sindresorhus/execa#env) would set all environment variables, but instead it only extends `process.env`. Setting [`extendEnv`](https://github.com/sindresorhus/execa#extendenv) resolved the problem